src/BrainPass/BrainPass.sol

```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
+import {Counters} from "openzeppelin-contracts/contracts/utils/Counters.sol";
+import {Owned} from "solmate/auth/Owned.sol";
+import {SafeMath} from "openzeppelin-contracts/contracts/utils/math/SafeMath.sol";
+
+interface IERC20 {
+    function transferFrom(
+        address from,
+        address to,
+        uint256 amount
+    ) external returns (bool);
+
+    function transfer(address to, uint256 amount) external returns (bool);
+
+    function balanceOf(address account) external view returns (uint256);
+}
+
+/// @title BRAIN Pass NFT
+/// @author Oleanji
+/// @notice A pass for IQ Wiki Editors
+
+contract BrainPassCollectibles is ERC721, Owned {
+    /// -----------------------------------------------------------------------
+    /// Errors
+    /// -----------------------------------------------------------------------
+
+    error MintingPaymentFailed();
+    error IncreseTimePaymentFailed();
+    error UserBalanceNotEnough();
+
+    /// -----------------------------------------------------------------------
+    ///  Inheritances
+    /// -----------------------------------------------------------------------
+    using SafeMath for uint256;
+    using Counters for Counters.Counter;
+
+    /// -----------------------------------------------------------------------
+    /// Structs
+    /// -----------------------------------------------------------------------
+    struct UserPassItem {
+        uint256 tokenId;
+        uint256 passId;
+        uint256 startTimestamp;
+        uint256 endTimestamp;
+    }
+
+    struct PassType {
+        uint256 passId;
+        string name;
+        uint256 pricePerDay;
+        string tokenURI;
+        uint256 maxTokens;
+        uint256 discount;
```

Comment:
whats maxTokens & discount?

Reply:
I think maxTokens should be the max no of NFTS that can be minted (in the case of a limited supply). and for the discount, we felt we might want to give a bit of discount to those subscribing for a long time  at once(say a yr subscriptions).

Reply:
you mention here that the discount param is "to those subscribing for a long time at once". Code doesnt do that at all. it just apply a generic discount to price. which its the same to not having that param and reduce the price per day.

Reply:
this has never been addressed. Please review my comments before asking for a new code review

Reply:
third time: please remove discount. it doesnt make any sense current implementation.

if you do a 10% discount its the same that if you just reduce  price by 10%.